Scripts/Runtime/Entities/Tasks/TaskFlow/TaskFlowGraph.cs

```diff
@@ -0,0 +1,399 @@
+        {
+            NodeLookup nodeLookup = GetOrCreateNodeLookup(taskSystem, null);
+            nodeLookup.Dispose();
+
+            JobNodeLookup jobNodeLookup = GetOrCreateJobNodeLookup(taskSystem, null);
+            jobNodeLookup.Dispose();
+
+            List<AbstractTaskDriver> taskDrivers = GetTaskDrivers(taskSystem);
+            taskDrivers.Clear();
+        }
+
+        public void DisposeFor(AbstractTaskSystem taskSystem, AbstractTaskDriver taskDriver)
+        {
+            NodeLookup nodeLookup = GetOrCreateNodeLookup(taskSystem, taskDriver);
+            nodeLookup.Dispose();
+
+            JobNodeLookup jobNodeLookup = GetOrCreateJobNodeLookup(taskSystem, taskDriver);
+            jobNodeLookup.Dispose();
+        }
+
+
+        //*************************************************************************************************************
+        // DATA STREAMS
+        //*************************************************************************************************************
+
+        public void CreateTaskStreams(AbstractTaskSystem taskSystem, AbstractTaskDriver taskDriver = null)
+        {
+            Debug_EnsureNotHardened();
+            RegisterTaskDriverToTaskSystem(taskSystem, taskDriver);
+
+            Type type;
+            object instance;
+            if (taskDriver == null)
+            {
+                type = taskSystem.GetType();
+                instance = taskSystem;
+            }
+            else
+            {
+                type = taskDriver.GetType();
+                instance = taskDriver;
+            }
+
+            //Get all the fields
+            FieldInfo[] systemFields = type.GetFields(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
+            foreach (FieldInfo systemField in systemFields)
+            {
+                if (!ABSTRACT_TASK_STREAM_TYPE.IsAssignableFrom(systemField.FieldType))
+                {
+                    continue;
+                }
+
+                IgnoreTaskStreamAttribute ignoreTaskStreamAttribute = systemField.GetCustomAttribute<IgnoreTaskStreamAttribute>();
+                if (ignoreTaskStreamAttribute != null)
+                {
+                    continue;
+                }
+
+                Debug_CheckFieldIsReadOnly(systemField);
+                Debug_CheckFieldTypeGenericTypeArguments(systemField.FieldType);
+
+                NodeLookup lookup = GetOrCreateNodeLookup(taskSystem, taskDriver);
+
+                //Get the data type 
+                AbstractTaskStream taskStream = TaskStreamFactory.Create(systemField.FieldType, systemField.FieldType.GenericTypeArguments[0]);
+
+                //Ensure the System's field is set to the task stream
+                systemField.SetValue(instance, taskStream);
+
+                //Create the node
+                DataStreamNode dataStreamNode = lookup.CreateDataStreamNode(taskStream, taskStream.GetDataStream());
```

Comment:
See below on line 115. The node isn't always the DataStream, it's sometimes the PendingCancelDataStream.

This will probably get reworked a bit in #68, #71 and or #63, I've added a TODO.